### PR TITLE
AG-13008, AG-13041 Fix Delete/Escape key deleting the annotation in text-edit mode

### PR DESCRIPTION
**Pull request by olegat**
https://ag-grid.atlassian.net/browse/AG-13008
https://ag-grid.atlassian.net/browse/AG-13041

### CODE REVIEW

File: packages/ag-charts-enterprise/src/features/annotations/annotationsStateMachine.ts [hunk 1]
```diff
@@ -390,8 +390,12 @@ export class AnnotationsStateMachine extends StateMachine<States, AnnotationType
 
                 delete: () => {
                     if (this.active == null) return;
+                    this.transition('cancel');
```
Comment (lsjroberts) [inline packages/ag-charts-enterprise/src/features/annotations/annotationsStateMachine.ts]: This is an anti-pattern, we should not transition inside these events. Partly as that could easily create an infinite loop, but also as it goes against the design of state machines.

**Review by lsjroberts — CHANGES_REQUESTED** [review]
There's been a lot of back and forth on deleting annotations due to changes in interactions.

Please make sure any changes continue to work for both of these fixes:

- https://github.com/ag-grid/ag-charts/pull/2160
- https://github.com/ag-grid/ag-charts/pull/2454

File: packages/ag-charts-website/e2e/toolbar.spec.ts [hunk 2]
```diff
@@ -126,4 +126,46 @@ test.describe('toolbar', () => {
         await page.click(SELECTORS.canvas, { position: { x: 100, y: 100 } });
         await expect(page).toHaveScreenshot('callout-6-change-fill-color.png', { animations: 'disabled' });
     });
+
+    test('AG-13008 delete annotation', async ({ page }) => {
```
Comment (lsjroberts) [inline packages/ag-charts-website/e2e/toolbar.spec.ts:140]: Also, can we just put some simple tests around this into the existing `test('text', ...` block? These tests are supposed to be just smoke tests.

File: packages/ag-charts-enterprise/src/features/annotations/annotationsStateMachine.ts (hunk 1 above)
Comment (olegat) [reply packages/ag-charts-enterprise/src/features/annotations/annotationsStateMachine.ts]: Fair. Another solution could be to store the `visible: boolean` state in the `TextInput` class and check that instead. I'll do that.

Comment (olegat) [reply packages/ag-charts-enterprise/src/features/annotations/annotationsStateMachine.ts]: Done.

File: packages/ag-charts-website/e2e/toolbar.spec.ts (hunk 2 above)
Comment (olegat) [reply packages/ag-charts-website/e2e/toolbar.spec.ts:140]: I've added tests to `test('text', ...)` for the <kbd>Backspace</kbd> and <kbd>Delete</kbd> keys

**Review by lsjroberts — APPROVED** [review]
LGTM, thanks!

**Review by alantreadway — APPROVED** [review]
LGTM

**Comment by olegat** [comment]
> There's been a lot of back and forth on deleting annotations due to changes in interactions.
> 
> Please make sure any changes continue to work for both of these fixes:
> 
> * [Ag 12476/annotations floating toolbar delete functionality #2160](https://github.com/ag-grid/ag-charts/pull/2160)
> * [AG-12598 Fix escape to cancel creating annotation #2454](https://github.com/ag-grid/ag-charts/pull/2454)

I've checked these.

✅ AG-12476 - No regression
✅ AG-12598 - No regression

**Comment by olegat** [comment]
Rebased and resolved conflict by dropping the e2e snapshot update to this file:
[packages/ag-charts-website/e2e/zoom.spec.ts-snapshots/zoom-crosshairs-after-wheel-zoom-chromium-linux.png](https://github.com/ag-grid/ag-charts/blob/28b713bb53099125953b647ef0827e89c9a39897/packages/ag-charts-website/e2e/zoom.spec.ts-snapshots/zoom-crosshairs-after-wheel-zoom-chromium-linux.png)

**Pull request merged**